### DISCUSSION TITLE
Question related to general direction of the static grid file

### DISCUSSION THREAD

**Discussion by chiaweh2** [NOAA-GFDL/CEFI-regional-MOM6#86] [Q&A] [answered]
Hi all, 

I am trying to set up a more scalable data structure on the CEFI data portal - OPENDAP access. I noticed that the static.nc file which contain the grid information would be different from region to region this is certain. But would this also be different among historical run, forecast, and prediction. Also would this be different between variables? Andrew mentioned ice would have its own static file. Would this be the same across regions? 

Thanks!

**Comment by theresa-cordero** [comment, marked as answer]
Hi Chia,

Yes, each region will have it's own static file but they should be consistent for the historical runs, forecasts, and predictions. While the sea ice and ocean may produce their own static files, we are running the ocean and the sea ice models on the same grid so the static files should be the consistent. There is some information that may be only in the ocean_static file such as the ocean depth, so I would suggest using the ocean static file rather than the sea ice.

For three dimensional data, the vertical grid data is not in the static file but within the data file. That might be something to keep in mind.

**Reply by chiaweh2** [reply]
@theresa-morrison thank you so much for the answer. This helps a lot on setting up the data structure and related processes on the data portal. Just want to also confirm that the rotation matrix will also be in the same place. Currently, I am processing the NWA ssu ssv as a starting point where I use the `ice_monthly.static.nc` `cosrot` and `sinrot` to rotating the vector field to true north and true east. Would this also be in the ice field for NEP for example?  And just an interesting discovery is that the geolon and geolat has mask (NaN) in the `ice_monthly.static.nc` file. It does not really affect the whole process but it is something I find it surprising.

**Comment by yichengt900** [comment]
@chiaweh2, regarding the velocity rotation for the ocean component (e.g., `ssu` and `ssv`), I suggest using the `angle_dx` variable (please be mindful of the unit, it should be rad) found in `ocean_hgrid.nc` to ensure the velocities are correctly aligned with the grid points. [Here](https://github.com/NOAA-GFDL/CEFI-regional-MOM6/blob/27cba7c152a7b0f24d9c83f2592a3e2cd4cb08eb/tools/boundary/boundary.py#L17) is a Python function we use to rotate ocean velocities from earth-relative to model-relative for [initial condition](https://github.com/NOAA-GFDL/CEFI-regional-MOM6/blob/27cba7c152a7b0f24d9c83f2592a3e2cd4cb08eb/tools/initial/write_glorys_initial.py#L96-L100) generation. In your case, you would apply the transpose (from model space to earth lat/lon):
```
    urot = np.cos(angle) * u - np.sin(angle) * v
    vrot = np.sin(angle) * u + np.cos(angle) * v
```

**Reply by chiaweh2** [reply]
@yichengt900 Thank you so much! However, I did not see this `ocean_hgrid.nc` in the output file currently. I am guessing this is a model input rather than model output? In this case, I have a more general question. Would it be wrong/inaccurate to use the `ice_monthly.static.nc`'s `cosrot` and `sinrot` to do the rotation? Since this is provided in the standard model run output in the ice static field, I think it make sense to use this instead of the model input files, IF AND ONLY IF the calculation is correct. Would love to know your opinion on this!

**Reply by theresa-cordero** [reply]
@chiaweh2 yes, `ocean_hgrid.nc` is an input file. One resaon we tend to use it instead of the `static.nc` files is that it does not have the masked values. I would hope that the calculation is the same using either file, but it would be worth confirming that before just using the output file.

**Reply by chiaweh2** [reply]
@theresa-morrison thanks! Could you point me to the `ocean_hgrid.nc` location? I am happy to do test since I already have the result derived from the ice_monthly.static.nc file.

**Reply by yichengt900** [reply]
@chiaweh2,

Another reason we use `ocean_hgrid.nc` is that it serves as a supergrid file, containing both `h` and `q` points, given that MOM6-SIS2 utilizes an Arakawa C-grid. You may have observed that the dimensions of `ssu` and `ssv` are slightly different (time, yh, xq vs. time, yq, xh). Additionally, the `COSRT` and `SINRT` fields in `ice_static.nc` are located at h (t) points, which may not exactly align with your `ssu` and `ssv`. Here is the ocean_hgrid.nc for the NWA domain for your reference:
`wget ftp.gfdl.noaa.gov:/pub/Yi-cheng.Teng/ocean_hgrid.nc`

**Reply by chiaweh2** [reply]
@yichengt900 Ah! yes, I make sure to interpolate to the tracer grid before doing the rotation. I can image to do accurate velocity field rotation on the naive stagger u v grid is better than the doing it on the tracer grid for many applications. But the purpose of the rotation I am doing is to actually have u v on the tracer grid which align with all other fields for the data portal visualization. But this is certainly a good thing to know and I will definitely make notes in some tutorial that I am making. Thanks for setting up a public share, I got the data!

**Reply by yichengt900** [reply]
> @yichengt900 Ah! yes, I make sure to interpolate to the tracer grid before doing the rotation. I can image to do accurate velocity field rotation on the naive stagger u v grid is better than the doing it on the tracer grid for many applications. But the purpose of the rotation I am doing is to actually have u v on the tracer grid which align with all other fields for the data portal visualization. But this is certainly a good thing to know and I will definitely make notes in some tutorial that I am making. Thanks for setting up a public share, I got the data!

@chiaweh2,

If you have already remapped u and v to the tracer grid points, then `COSRT` and `SINRT` should likely be sufficient for your application. However, it's also a good practice to double-check the results using ocean_hgrid.nc as a sanity check.
